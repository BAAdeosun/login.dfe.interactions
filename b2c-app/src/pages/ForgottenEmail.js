import React from 'react';
import components from '../components';
import { onChange, onError } from '../helpers/pageUpdatesHandler';

class ForgottenEmail extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            firstName: null,
            lastName: null,
            day: null,
            month: null,
            year: null,
            postcode: null,
            showErrors: false,
            errors: []
        }
        this.handleSubmit = this.handleSubmit.bind(this);
        this.onError = onError.bind(this);
        this.onChange = onChange.bind(this);
    }

    componentDidMount() {
        document.getElementById('api').style.display = 'none';
        document.title = 'Find your email address | National Careers Service';
    }

    handleSubmit(e) {
        e.preventDefault();
        //update error messages
        this.state.errors.forEach((error) => {
            error.visible.text = error.current.text;
            error.visible.showSummaryText = error.current.showSummaryText;
        });
        //do something to validate and decide if we submit or show errors
        if (this.state.firstName &&
            this.state.lastName &&
            this.state.day &&
            this.state.month &&
            this.state.year &&
            this.state.postcode) {
            this.setState({ showErrors: false });
            //everything is valid, set data and submit B2C form
            this.setDataAndSubmit();
        }
        else {
            //show errors in each component
            this.setState({ showErrors: true });
        }
    }

    setDataAndSubmit() {
        //retrieve all elements we will need and set their values
        let b2cFirstName = document.getElementById('givenName');
        let b2cLastName = document.getElementById('surname');
        let b2cDobDay = document.getElementById('day');
        let b2cDobMonth = document.getElementById('month');
        let b2cDobYear = document.getElementById('year');
        let b2cPostcode = document.getElementById('postCode');
        let b2cSubmitButton = document.getElementById('continue');

        if (b2cFirstName && b2cLastName &&
            b2cDobYear && b2cDobMonth && b2cDobYear &&
            b2cPostcode &&
            b2cSubmitButton) {
            b2cFirstName.value = this.state.firstName;
            b2cLastName.value = this.state.lastName;
            b2cDobDay.value = this.state.day;
            b2cDobMonth.value = this.state.month;
            b2cDobYear.value = this.state.year;
            b2cPostcode.value = this.state.postcode;
            //submit B2C form
            b2cSubmitButton.click();
        }
    }

    render() {
        return (
            <div id="forgottenEmail">

                <div className="govuk-width-container">
                    <components.Breadcrumbs />

                    <components.PageLevelErrorContainer errorItems={this.state.errors} />

                    <main className="govuk-main-wrapper">
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-two-thirds">
                                <components.PageTitle size='xl' title="Find your email address" />

                                <form id="activateAccountForm" onSubmit={this.handleSubmit} noValidate>
                                    <components.InputField
                                        inputId='firstName'
                                        inputLabel='First name'
                                        errorMessagePlaceholder='first name'
                                        onChange={this.onChange}
                                        onError={this.onError}
                                        showErrors={this.state.showErrors}
                                    />
                                    <components.InputField
                                        inputId='lastName'
                                        inputLabel='Last name'
                                        errorMessagePlaceholder='last name'
                                        onChange={this.onChange}
                                        onError={this.onError}
                                        showErrors={this.state.showErrors}
                                    />
                                    <components.DateOfBirth onChange={this.onChange} onError={this.onError} showErrors={this.state.showErrors} />
                                    <components.Postcode onChange={this.onChange} onError={this.onError} showErrors={this.state.showErrors} />
                                    <button className="govuk-button" id="preSubmit" type="submit">Find email address</button>
                                </form>
                            </div>
                        </div>
                    </main>

                </div>

            </div>
        )
    }
}

export default ForgottenEmail;
