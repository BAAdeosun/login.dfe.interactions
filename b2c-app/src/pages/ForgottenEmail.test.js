
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ForgottenEmail from './ForgottenEmail';

it('renders without crashing', () => {
    shallow(<ForgottenEmail />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<ForgottenEmail />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});