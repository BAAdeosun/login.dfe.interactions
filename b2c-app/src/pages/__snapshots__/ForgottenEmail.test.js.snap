// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="forgottenEmail"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Find your email address
            </h1>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="firstNameCustom"
                  >
                    First name
                  </label>
                  <input
                    className="govuk-input govuk-!-width-one-half"
                    id="firstNameCustom"
                    name="firstName"
                    noValidate={true}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="lastNameCustom"
                  >
                    Last name
                  </label>
                  <input
                    className="govuk-input govuk-!-width-one-half"
                    id="lastNameCustom"
                    name="lastName"
                    noValidate={true}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
                <div
                  className="govuk-form-group "
                >
                  <fieldset
                    aria-describedby="date-of-birth-hint"
                    className="govuk-fieldset"
                    id="dobFieldset"
                    role="group"
                  >
                    <legend
                      className="govuk-fieldset__legend govuk-fieldset__legend--l"
                    >
                      <label
                        className="govuk-label"
                      >
                        Date of birth
                      </label>
                    </legend>
                    <span
                      className="govuk-hint"
                      id="date-of-birth-hint"
                    >
                      For example, 31 3 1980
                    </span>
                    <div
                      className="govuk-date-input"
                      id="date-of-birth"
                    >
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobDay"
                          >
                            Day
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-2"
                            id="dobDay"
                            inputMode="numeric"
                            name="dobDay"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobMonth"
                          >
                            Month
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-2"
                            id="dobMonth"
                            inputMode="numeric"
                            name="dobMonth"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobYear"
                          >
                            Year
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-4"
                            id="dobYear"
                            inputMode="numeric"
                            name="dobYear"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                    </div>
                  </fieldset>
                </div>
                <div
                  className="govuk-form-group "
                >
                  <label
                    className="govuk-label"
                    htmlFor="postcodeCustom"
                  >
                    Enter your postcode
                  </label>
                  <span
                    className="govuk-hint"
                    id="postcode-hint"
                  >
                    For example, SW1A 1AA
                  </span>
                  <input
                    className="govuk-input govuk-input--width-10"
                    id="postcodeCustom"
                    name="postcode"
                    noValidate={true}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Find email address
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
